airbyte-workers/src/main/java/io/airbyte/workers/DefaultReplicationWorker.java

```diff
@@ -134,7 +120,10 @@ public StandardSyncOutput run(StandardSyncInput syncInput, Path jobRoot) throws
       throw new WorkerException("Sync was cancelled.");
```

Comment:
Are we really satisfied of throwing an exception when it is canceled? (it is not an error)

Reply:
no, i think you're right. i got to massage (write tests for) these failure cases so that i can ensure that the lifecycle make sense.